### ISSUE TITLE
Debugger stops at breakpoint in a loop only once (Windows Arm64)

### ISSUE THREAD

**Issue by sergunya** [go-delve/delve#3183]
**1. dlv version**
```
Delve Debugger
Version: 1.9.1
Build: $Id: d81b9fd12bfa603f3cf7a4bc842398bd61c42940 $
```

**2. go version**
`go version go1.19.3 windows/arm64`

**3. Operating system and processor architecture**
```
Edition	       Windows 11 Pro
Version	       22H2
OS build	       22621.521
System type  64-bit operating system, ARM-based processor
```

**6. Steps to reproduce**
   - Install delve for Windows arm64 `go install -tags exp.winarm64 github.com/go-delve/delve/cmd/dlv`
   - Create sample.go file with the outlined below content
      `package main

        import (
            "fmt"
        )

        func main() {
            length, capacity, y := 1000, 2000, 3
            AllResults := make([]int, length, capacity)
            for i := 0; i < length; i++ {
                result := i / y // Hi there
                fmt.Println(result)
                AllResults[i] = result
            }
            j := 0
            fmt.Println(j)
        }`
   - Start debug client: `dlv debug sample.go`
   - Add breakpoint inside loop: `(dlv) break sample.go:12` 
   - Continue debug session  `(dlv) continue` (debugger stops at breakpoint)
   - Continue debug session further `(dlv) continue`  

**7. Expected result**
The debugger stops at the breakpoint in every cycle in a loop.

**8. Actual result**
The debugger stops at the breakpoint only in first cycle of the loop.

![example](https://user-images.githubusercontent.com/1156287/200295473-7577caca-f10a-4d2f-939e-3c97af5113ac.gif)

**Comment by aarzilli** [comment]
cc @qmuntal

**Comment by sergunya** [comment]
Also `step` action doesn't work. Should I create a separate issue or we can continue with this ticket?

**Comment by aarzilli** [comment]
I think it's fine to continue here.

**Comment by qmuntal** [comment]
Thanks for reporting this issue. I'll investigate it next week.

**Comment by qmuntal** [comment]
Which version of go are you using @sergunya? Can't reproduce this issue with `go version go1.19.2 windows/arm64`.

**Comment by sergunya** [comment]
@qmuntal hey, 
`go version go1.19.3 windows/arm64`

**Comment by qmuntal** [comment]
I'm afraid I can't reproduce this locally. I do have an OS build slightly newer than yours (`22621.819`), but I doubt it matters here.

**Comment by soerennielsen** [comment]
Delve version 1.24 (with exp. flag set), Go 1.23, Windows 11 24H2 Arm64. 

Time has passed, now it works excellent, stops in the loop everytime. Tested in vs code, not command line though. 

Thanks for the install command line :-)
